pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java

```diff
@@ -308,6 +361,62 @@ protected synchronized boolean trySendMessagesToConsumers(ReadType readType, Lis
             }
         }
 
+        // Update the last sent position and remove ranges from individuallySentPositions if necessary
+        if (!allowOutOfOrderDelivery && lastSentPosition != null) {
```

Comment:
Just mark this for easier reading: update `lastSentPosition` to the position of the first consecutive sent message